### DISCUSSION TITLE
Is it safe to expose the user UUID generated by authentication to other users?

### DISCUSSION THREAD

**Discussion by calmdaysamuel** [supabase/supabase#3568] [Questions] [answered]
Is it safe to expose the user UUID generated by authentication to other users? Can hackers use it to bypass RLS that requires authentication or RLS that requires a user to by the owner of a row?

**Comment by dshukertjr** [comment, marked as answer]
Hi @calmdaysamuel!

It is totally safe to expose the UUID of authenticated user and it is totally expected. If you follow the [Row Level Security guide](https://supabase.io/docs/guides/auth/row-level-security) and properly implement security policy, there's not much hackers can do with just user id of a user. 

If you would like to learn more, [Auth Deep Dive series](https://supabase.io/docs/learn/auth-deep-dive/auth-deep-dive-jwts) is a great place to learn why it is safe to expose certain information in Supabase such as Anon key or user id.
